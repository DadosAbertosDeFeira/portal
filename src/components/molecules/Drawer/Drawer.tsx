import classNames from "classnames";
import type { ComponentPropsWithRef } from "react";
import React, { forwardRef } from "react";

export type DrawerProps = ComponentPropsWithRef<"div"> & {
  isOpen: boolean;
  direction?: "left" | "right";
};

export const Drawer = forwardRef<HTMLDivElement, DrawerProps>(function Drawer(
  { isOpen, direction = "left", children, ...props },
  ref
) {
  const directions = {
    left: {
      closed: "left-[-100vw]",
      open: "left-0",
    },

    right: {
      closed: "right-[-100vw]",
      open: "right-0",
    },
  };

  return (
    <div
      className={classNames(
        directions[direction][isOpen ? "open" : "closed"],
        "z-50 border-red fixed w-[70vw] max-w-[325px] min-w-[280px] inset-y-0 bg-white transition-all duration-300 ease-in-out shadow-3"
      )}
      role="navigation"
      {...props}
      ref={ref}
    >
      {children}
    </div>
  );
});
