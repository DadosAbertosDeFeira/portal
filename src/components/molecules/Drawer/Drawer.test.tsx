import { render } from "@testing-library/react";

import { Drawer, type DrawerProps } from "./Drawer";

describe("Drawer", () => {
  const makeSut = ({ isOpen = false, ...props }: Partial<DrawerProps>) => {
    return render(<Drawer isOpen={isOpen} {...props} />);
  };

  // eslint-disable-next-line jest/expect-expect
  it("renders correctly", () => {
    const { asFragment } = makeSut({});

    expect(asFragment).toMatchSnapshot();
  });

  // eslint-disable-next-line jest/expect-expect
  it("renders open correctly", () => {
    const { asFragment } = makeSut({ isOpen: true });

    expect(asFragment).toMatchSnapshot();
  });
});
