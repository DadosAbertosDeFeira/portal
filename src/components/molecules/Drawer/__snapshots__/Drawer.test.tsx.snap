// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer renders correctly 1`] = `[Function]`;

exports[`Drawer renders open correctly 1`] = `[Function]`;
